src/markdown/block/blockquote.rs

```diff
@@ -0,0 +1,86 @@
+use super::Block;
+use super::Block::Blockquote;
+use crate::markdown::block::parse_blocks;
+
+pub fn parse_blockquote(lines: &[&str]) -> Option<(Block, usize)> {
```

Comment:
This shouldn't ever see multiple lines. With that assumption out of the way, the function logic should be vastly simplified.